electrumsv/tests/test_wallet.py

```diff
@@ -550,64 +619,169 @@ def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorag
+    if password_token_before is not None:
+        test_state.count_of_password_token_reencryptions += 1
+
+    # Derivation data in the KeyInstances table. Only for ImportedPrivkeyAccount types
+    keyinstance_private_keys: dict[str, str] = {}
     if "imported" == expected_type and "privkey" in wallet_filename:
         assert len(wallet.get_accounts()) == 2
         private_key_account = cast(ImportedPrivkeyAccount,
             [ entry for entry in wallet.get_accounts() if not entry.is_petty_cash() ][0])
         private_key_keystore = cast(Imported_KeyStore, private_key_account.get_keystore())
         # Pre-decrypt the prv for later comparison so the initial password is not needed there.
         for public_key, encrypted_prv in private_key_keystore._keypairs.items():
-            prv_keypairs[public_key.to_hex()] = pw_decode(encrypted_prv, initial_password)
-
-    password = "654321"
-    future, update_completion_event = wallet.update_password(initial_password, password)
+            keyinstance_private_keys[public_key.to_hex()] = pw_decode(encrypted_prv,
+                initial_password)
+            test_state.count_of_keyinstance_privkey_reencryptions += 1
+
+    # Derivation data in the MasterKeys table.
+    master_key_xprvs: list[str] = []
+    master_key_seeds: list[str] = []
+    master_key_passphrases: list[str] = []
+    for keystore in wallet.get_keystores():
+        data = keystore.to_derivation_data()
+        for entry_name in ("seed", "passphrase", "xprv"):
+            entry_value = cast(str | None, data.get(entry_name, None))
+            if not entry_value:
+                continue
+            decoded_value = pw_decode(entry_value, initial_password)
+            if entry_name == "seed":
+                master_key_seeds.append(decoded_value)
+                test_state.count_of_masterkeys_table_seed_reencryptions += 1
+            if entry_name == "xprv":
+                master_key_xprvs.append(decoded_value)
+                test_state.count_of_masterkeys_table_xprv_reencryptions += 1
+            if entry_name == "passphrase":
+                master_key_passphrases.append(decoded_value)
+                test_state.count_of_masterkeys_table_passphrase_reencryptions += 1
+
+    # Newer wallets have a `Servers` table with an `encrypted_api_key` column
+    encrypted_api_keys = list[str]()
+    for server in wallet.data.read_network_servers():
+        if server.encrypted_api_key is not None:
+            test_state.count_of_server_api_key_reencryptions += 1
+            decoded_value = pw_decode(server.encrypted_api_key, initial_password)
+            encrypted_api_keys.append(decoded_value)
+
+    new_password = "654321"
+    future, update_completion_event = wallet.update_password(initial_password, new_password)
     # Wait for the database update to finish.
     future.result(5)
     # Wait for the done callback to finish.
     update_completion_event.wait()
+    wallet.stop()
```

Comment:
So, we update the password then we throw away the cached state and do not check it is correct? The whole point of you doing this is that we gain an overview that we have encountered different encrypted data and that changing the password worked successfully and you are not even checking?